### PR TITLE
Fixes #1342: Configure WordpressCrawlerService with BaseScraper and Post

### LINKED ISSUE #1342: Configure WordpressCrawlerService with BaseScraper and Post
- Issue type: Feature request 

Task:

1) Configure WordpressCrawlerService with BaseScraper
2) Configure WordpressCrawlerService with Post

### PR DESCRIPTION
**Pull request by vibhcool**
Fixes #1342 

I have:
- [x] There is a corresponding issue for this pull request.
- [x] Mentioned the Issue number in the pull request commit message `Fixes #<number> commit message`
- [x] There is only strictly only one commit per issue.

### For the reviewers
I have:
- [ ] Reviewed this pull request by an authorized contributor.
- [ ] The reviewer is assigned to the pull request.

### CODE REVIEW

**Comment by vibhcool** [comment]
open for review
@Achint08 @djmgit @hemantjadon @kavithaenair @SKrPl @singhpratyush

File: src/org/loklak/api/search/WordpressCrawlerService.java [hunk 1]
```diff
@@ -57,87 +72,90 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
...
+
+   protected Post scrape(BufferedReader br, String type, String url) {
+        Post typeArray = new Post(true);
+        typeArray.put(this.scraperName, this.crawlWordpress(this.query, br).toArray());
+        return typeArray;
+    }
+
+
+    public Timeline2 crawlWordpress(String blogURL, BufferedReader br) {
+
```
Comment (djmgit) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: I know it was the beginning, but please this empty line :)

File: src/org/loklak/api/search/WordpressCrawlerService.java [hunk 2]
```diff
@@ -57,87 +72,90 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
...
+		        blogPostList[iterator][2] = author;
+	        }
+            // Content of article
+	        articleList_content = article.getElementsByClass("entry-content");
+	        for (Element blogs : articleList_content) {
+		        blogPostList[iterator][3] = blogs.text().toString();
+	        }
+
+	        iterator++;
+
```
Comment (djmgit) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: Here too

File: src/org/loklak/api/search/WordpressCrawlerService.java [hunk 3]
```diff
@@ -57,87 +72,90 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
...
+	        blogPost.put("blog_url", blogURL);
+	        blogPost.put("title", blogPostList[k][0]);
+	        blogPost.put("posted_on", blogPostList[k][1]);
+	        blogPost.put("author", blogPostList[k][2]);
+	        blogPost.put("content", blogPostList[k][3]);
+	        blogList.addPost(blogPost);
+        }
+
+        return blogList;
+
```
Comment (djmgit) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: Here too

File: src/org/loklak/api/search/WordpressCrawlerService.java [hunk 4]
```diff
@@ -57,87 +72,90 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
...
+            this.query  = this.getExtraValue("url");
+        }
+    }
+
+   protected Post scrape(BufferedReader br, String type, String url) {
+        Post typeArray = new Post(true);
+        typeArray.put(this.scraperName, this.crawlWordpress(this.query, br).toArray());
+        return typeArray;
+    }
+
```
Comment (djmgit) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: remove one empty line here

File: src/org/loklak/api/search/WordpressCrawlerService.java [hunk 5]
```diff
@@ -57,87 +72,90 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
...
+        Timeline2 blogList = new Timeline2(this.order);
+
+        try {
+	        blogHTML = Jsoup.parse(bufferedReaderToString(br));
+        } catch (IOException e) {
+	        DAO.trace(e);
+        }
+
+        String[][] blogPostList = new String[100][4];
```
Comment (Achint08) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: Why `[100]`?

Comment (djmgit) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: It would be nice if the size is set dynamically :)

Comment (vibhcool) [inline src/org/loklak/api/search/WordpressCrawlerService.java]: I just configured BaseScraper and Post. this isn't related to PR. I will fix it.

**Comment by vibhcool** [comment]
@Achint08 @djmgit please re- review

**Comment by mariobehling** [comment]
Please resolve conflicts.

**Comment by vibhcool** [comment]
@mariobehling , rebased and resolved merge conflicts :)

**Pull request merged**